### ISSUE TITLE
help command still prone to SQL shenanigans

### ISSUE THREAD

**Issue by jalles76** [luciensadi/AwakeMUD#2]
help 'hunger%' OR '%status

Program received signal SIGSEGV, Segmentation fault.
0x00007ffff6acc319 in _IO_vfprintf_internal (s=<optimized out>, format=<optimized out>, ap=<optimized out>) at vfprintf.c:2047
2047    vfprintf.c: No such file or directory.
(gdb) bt
#0  0x00007ffff6acc319 in _IO_vfprintf_internal (s=<optimized out>, format=<optimized out>, ap=<optimized out>) at vfprintf.c:2047
#1  0x00007ffff6acee71 in buffered_vfprintf (s=s@entry=0x7ffff6e421c0 <_IO_2_1_stderr_>, format=format@entry=0x7fffffff5a90 "Executing query: SELECT * FROM help_topic WHERE name LIKE '%\\'hunger%\\' OR \\'%status%' ORDER BY name ASC", args=args@entry=0x7fffffff5998) at vfprintf.c:2356
#2  0x00007ffff6ac9dfe in _IO_vfprintf_internal (s=0x7ffff6e421c0 <_IO_2_1_stderr_>, format=0x7fffffff5a90 "Executing query: SELECT * FROM help_topic WHERE name LIKE '%\\'hunger%\\' OR \\'%status%' ORDER BY name ASC", ap=0x7fffffff5998) at vfprintf.c:1313
#3  0x00000000005bacd9 in log (format=0x7fffffff5a90 "Executing query: SELECT * FROM help_topic WHERE name LIKE '%\\'hunger%\\' OR \\'%status%' ORDER BY name ASC") at utils.cpp:633
#4  0x000000000053891e in mysql_wrapper (mysql=0x89c580, query=0x7fffffff7ae0 "SELECT * FROM help_topic WHERE name LIKE '%\\'hunger%\\' OR \\'%status%' ORDER BY name ASC") at newdb.cpp:54
#5  0x0000000000433381 in display_help (help=0x7fffffff9b30 "\200\241\377\377\377\177", arg=0x7fffffffddb5 "'hunger%' OR '%status") at act.informative.cpp:2239
#6  0x00000000004336bb in do_help (ch=0xb41630, argument=0x7fffffffddb5 "'hunger%' OR '%status", cmd=200, subcmd=0) at act.informative.cpp:2278
#7  0x0000000000518a1d in command_interpreter (ch=0xb41630, argument=0x7fffffffddb0 "help 'hunger%' OR '%status", tcname=0xb41ea0 "Finster") at interpreter.cpp:1333
#8  0x00000000004aca2a in game_loop (mother_desc=3) at comm.cpp:660
#9  0x00000000004abbfe in init_game (port=4000) at comm.cpp:351
#10 0x00000000004ab669 in main (argc=1, argv=0x7fffffffe708) at comm.cpp:235
(gdb)

**Comment by luciensadi** [comment]
Thanks for providing the stack trace. I did some digging, and this looks like it's actually a problem with passing the string to the format function. Check this out:

> help %s %d %f %ld
> No such help file exists.

produces in console

> May 25 06:33:54 :: Executing query: SELECT * FROM help_topic WHERE name LIKE '%s 1606389179 0.000000 1407347997671763777753757666454RDER BY name ASC

I've moved the old log function to log_vfprintf and added a new log function that does not format the string passed to it. This should make it easier to avoid leaking memory into the logs.
